src/connector/src/sink/deltalake/mod.rs

```diff
@@ -0,0 +1,187 @@
+pub struct DeltaLakeCommon {
+    #[serde(rename = "location")]
+    pub location: String,
+    #[serde(flatten)]
+    pub aws_auth_props: AwsAuthProps,
+
+    #[serde(rename = "gcs.service.account")]
+    pub gcs_service_account: Option<String>,
+    /// Commit every n(>0) checkpoints, default is 10.
+    #[serde(default = "default_commit_checkpoint_interval")]
+    #[serde_as(as = "DisplayFromStr")]
+    pub commit_checkpoint_interval: u64,
+}
+
+impl EnforceSecret for DeltaLakeCommon {
+    const ENFORCE_SECRET_PROPERTIES: Set<&'static str> = phf_set! {
+        "gcs.service.account",
+    };
+
+    fn enforce_one(prop: &str) -> crate::error::ConnectorResult<()> {
+        AwsAuthProps::enforce_one(prop)?;
+        if Self::ENFORCE_SECRET_PROPERTIES.contains(prop) {
+            return Err(EnforceSecretError {
+                key: prop.to_owned(),
+            }
+            .into());
+        }
+
+        Ok(())
+    }
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, WithOptions)]
+pub struct DeltaLakeConfig {
+    #[serde(flatten)]
+    pub common: DeltaLakeCommon,
+
+    pub r#type: String,
+}
+
+impl EnforceSecret for DeltaLakeConfig {
+    fn enforce_one(prop: &str) -> crate::error::ConnectorResult<()> {
+        DeltaLakeCommon::enforce_one(prop)
+    }
+}
+
+impl DeltaLakeConfig {
+    pub fn from_btreemap(properties: BTreeMap<String, String>) -> Result<Self> {
+        let config = serde_json::from_value::<DeltaLakeConfig>(
+            serde_json::to_value(properties).map_err(|e| SinkError::DeltaLake(e.into()))?,
+        )
+        .map_err(|e| SinkError::Config(anyhow!(e)))?;
+        Ok(config)
+    }
+}
+
+#[derive(Debug)]
+pub struct DeltaLakeSink {
+    pub config: DeltaLakeConfig,
+    #[cfg_attr(not(feature = "sink-deltalake"), allow(dead_code))]
+    param: SinkParam,
+}
+
+impl EnforceSecret for DeltaLakeSink {
+    fn enforce_secret<'a>(
+        prop_iter: impl Iterator<Item = &'a str>,
+    ) -> crate::error::ConnectorResult<()> {
+        for prop in prop_iter {
+            DeltaLakeCommon::enforce_one(prop)?;
+        }
+        Ok(())
+    }
+}
+
+impl DeltaLakeSink {
+    pub fn new(config: DeltaLakeConfig, param: SinkParam) -> Result<Self> {
+        Ok(Self { config, param })
+    }
+}
+
+/// Dummy implementation
+#[cfg(not(feature = "sink-deltalake"))]
+mod dummy {
+
+    use std::collections::BTreeMap;
+
+    use anyhow::anyhow;
+    use sea_orm::DatabaseConnection;
+    use tokio::sync::mpsc::UnboundedSender;
+
+    use super::*;
+    use crate::connector_common::IcebergCompactionStat;
+    use crate::sink::DummySinkCommitCoordinator;
+    use crate::sink::trivial::{TrivialSink, TrivialSinkName};
+
+    const ERROR_MESSAGE: &str = "RisingWave is not compiled with feature `sink-deltalake`";
+
+    #[derive(Debug)]
+    pub struct DeltaLakeSinkName;
+
+    impl TrivialSinkName for DeltaLakeSinkName {
+        const SINK_NAME: &'static str = DELTALAKE_SINK;
+    }
+
+    impl Sink for DeltaLakeSink {
```

Comment:
Shall we make this dummy sink implementation to be more general so that in the future when we want to support optionally disabling more sinks, we can easily reuse this dummy impl?